lib/top/compat_top.ml

```diff
@@ -1,52 +1,26 @@
-open Mdx.Compat
-
-#if OCAML_VERSION >= (4, 8, 0)
-let try_finally ~always f = Misc.try_finally f ~always
-#else
-let try_finally ~always f = Misc.try_finally f always
-#endif
-
 let map_error_loc ~f (error : Location.error) =
```

Comment:
It looks like a bunch of functions in this module don't have `#if`s anymore. Can we remove them from this compat module? Usually it works well to inline them at the call site, or at least we can move the definition to the main library code otherwise.

Reply:
Good point, this removes even more functions (especially decreasing the size of the exposed interface). For the most part I kept the (non-exposed) helper functions since given they have reasonably useful names I found it more descriptive to keep them as such.